test-data/unit/check-attr.test

```diff
@@ -705,3 +705,69 @@ import attr
 class A(object):
     x = attr.ib()
 [builtins_py2 fixtures/bool.pyi]
+
+[case testAttrsFrozenSubclass]
+import attr
+
+@attr.dataclass
+class NonFrozenBase:
+    a: int
+
+@attr.dataclass(frozen=True)
+class FrozenBase:
+    a: int
+
+@attr.dataclass(frozen=True)
+class FrozenNonFrozen(NonFrozenBase):
+    b: int
+
+@attr.dataclass(frozen=True)
+class FrozenFrozen(FrozenBase):
+    b: int
+
+@attr.dataclass
+class NonFrozenFrozen(FrozenBase):
+    b: int
+
+# Make sure these are untouched
+non_frozen_base = NonFrozenBase(1)
+non_frozen_base.a = 17
+frozen_base = FrozenBase(1)
+frozen_base.a = 17  # E: Property "a" defined in "FrozenBase" is read-only
+
+a = FrozenNonFrozen(1, 2)
+a.a = 17  # E: Property "a" defined in "FrozenNonFrozen" is read-only
```

Comment:
This error message is a bit weird because `a` is defined in `NonFrozenBase`, _not_ in `FrozenNonFrozen`. My understanding is that this is `attrs` semantics to "redefine" all parent class attributes as read-only if the class is frozen.

If you think this error message can be somehow easily improved, then could you please do this today? (I am going to cut the release branch today evening or tomorrow morning.) Otherwise, the error message can be fixed in a separate PR (please create an issue in this case).

Reply:
The `attrs` semantics actually add a `__setattr__` that raises to the frozen class.  But mypy doesn't seem to handle that.  

As for the message I think it's an existing problem:
```
class C:
    @property
    def foo(self) -> int:
        return 17
class D(C):
    pass
d = D()
d.foo = 18  # error: Property "foo" defined in "D" is read-only
```

PS. Another way to make frozen better would be to implement #4572

Reply:
OK, I opened a separate issue for error message.

As for `__setattr__`, I think we shouldn't wait, this PR already correctly fixes the crash, so I am going to merge it now.